tsl/src/compression/create.c

```diff
@@ -158,11 +191,19 @@ build_columndefs(CompressionSettings *settings, Oid src_relid)
 		 * Put the metadata columns before the compressed columns, because they
 		 * are accessed before decompression.
 		 */
-		if (is_orderby)
+		if (is_segmentby)
+		{
+			/* No additional metadata for segmentby. */
+		}
```

Comment:
Do we really need this?

Reply:
The rest of conditions have to be under `!is_segmentby`, so I wrote it this way to decrease indentation :) Looks a little weird I agree, will switch it to the more traditional form.